### DISCUSSION TITLE
Azure, EMQX 5, operator 2.1.0 and StorageClass - unclear documentation

### DISCUSSION THREAD

**Discussion by klemen-df** [emqx/emqx-operator#612] [Q&A] [answered]
I'll write how I wen't about it :

I was looking for a way to deploy EMQX 5 to Azure Kubernetes service and found [EMXQ Operator](https://github.com/emqx/emqx-operator), great.

Went on to read '[Getting Started](https://github.com/emqx/emqx-operator/blob/main/docs/en_US/getting-started/getting-started.md)', deployed with the example [stated here](https://github.com/emqx/emqx-operator/blob/main/docs/en_US/getting-started/getting-started.md#deploy-emqx-5), great.

It deploys and it seems that it works.

Then, searching for something else, found this ; '[Deployment on Public Cloud - Azure AKS](https://docs.emqx.com/en/emqx-operator/latest/deployment/azure-deployment.html#deploy-emqx-clusters-on-azure-aks-using-emqx-operator)'.

Here it says to first create a storageClass 'azurefile-csi-nfs'. Also, the example is with emqx-ee and with an older version of the operator (apiversion of v1beta3) and an older version of emqx (4.4.8).

So, what am I missing here :
- example for emqx operator (instead of emqx-enterprise) (probably just replacing emqx-ee with emqx)
- a line of text or two saying how does not creating the storageClass effect the emqx (operator)
- example with the latest operator (apiversion v2alpha1), emqx >= 5

**Comment by Rory-Z** [comment, marked as answer]
Hi, @KlemenDanfoss thanks for your feedback, we will update documents soon.
Let's back to your concerns:

> example for emqx operator (instead of emqx-enterprise) (probably just replacing emqx-ee with emqx)

Yes, just use `kind: EmqxBroker` replace of `kind: EmqxEnterprise`, and it will work

> a line of text or two saying how does not creating the storageClass effect the emqx (operator)

StorageClass holds EMQX runtime data, such as configuration changes made via the EMQX Dashboard, etc. If storageClass is not enabled, then when the EMQX CR is removed and redeployed, you will get a completely new cluster.
More info, please check https://docs.emqx.com/en/emqx-operator/latest/tasks/configure-emqx-persistence.html

> example with the latest operator (apiversion v2alpha1), emqx >= 5

Please check https://github.com/emqx/emqx-operator/blob/main/config/samples/emqx/v2alpha1/emqx-full.yaml, and change the `.spec.coreTemplate.spec.volumeClaimTemplates.storageClassName` to Azure's storeClass name
